### PR TITLE
Refactor spells to use regular damage pipeline

### PR DESCRIPTION
**Pull request by CarlosFdez**
Enables the ability to add damage dice to spells. Also sneaky fix so that elite only applies to the first instance of damage.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
DynamicBaseDamageData needs renaming. Spell isn't quite right. Its not a string either and its not from a formula. BaseTermsDamageData?

File: src/module/system/damage/types.ts [hunk 1]
```diff
@@ -55,18 +55,29 @@ interface DamageRollContext extends BaseRollContext {
 }
 
 interface DamageFormulaData {
-    base: BasicDamageData;
+    base: BaseDamageData | BaseDamageData[];
```
Comment (stwlam) [inline src/module/system/damage/types.ts:58]: Just make it an array from everywhere

Comment (CarlosFdez) [reply src/module/system/damage/types.ts:58]: Sure. It'll blow up the lines of code but we can do that.

**Comment by CarlosFdez** [comment]
In retrospect I'm considering renaming combineTerms() to simplifyFormula(). That way it'll be more distinguished from combineSimpleTerms() which just combines terms into a new list of combined terms.

**Pull request merged**